google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -58,6 +82,13 @@ private Timestamp(long seconds, int nanos) {
     this.nanos = nanos;
   }
 
+  public static Timestamp ofTimeMicroseconds(long microseconds) {
```

Comment:
Because of the existing `ofTimeSecondsAndNanos`. I can go ahead and remove `Time` from both.